core/trino-main/src/main/java/io/trino/sql/planner/PlanFragmenter.java

```diff
@@ -323,9 +323,6 @@ public PlanNode visitRefreshMaterializedView(RefreshMaterializedViewNode node, R
         @Override
         public PlanNode visitTableWriter(TableWriterNode node, RewriteContext<FragmentProperties> context)
         {
-            if (node.getPartitioningScheme().isPresent()) {
```

Comment:
The `Fragmenter` has to choose the distribution of a stage. Since the partitioning handle of an exchange is getting changed from `FIXED_HASH_DISTRIBUTION` to `SCALED_WRITER_HASH_DISTRIBUTION` (and similar for connector specific partitioning) the `Fragmenter` becomes confused, as it sees different partitioning handles in `TableWriterNode` and `ExchangeNode`.

I honestly don't know what's the best way to approach this. From one side I don't feel like taking into account the handle in `TableWriterNode` is necessary at this point, as distribution should generally be determined by the `ExchangeNode`s / `TableScan`. If an incorrect exchange is inserted - plan is broken anyway. 

However it still might be a good "sanity" check. But to keep it functional we would have to overwrite the partitioning in `TableWriteNode` what would mean modifying the `TableWriterNode` at the `AddExchanges` step what also seems a little weird, especially since the `AddLocalExchange` may not expect it. 

@sopel39 @gaurav8297 Thoughts?